spec/furniture/marketplace/cart_products_controller_request_spec.rb

```diff
@@ -5,36 +5,117 @@
   let(:space) { marketplace.space }
   let(:room) { marketplace.room }
   let(:member) { create(:membership, space: space).member }
+  let(:document_root_element) { Nokogiri::HTML::Document.parse(response.body) }
+  let(:request_as) { :html }
 
   describe "#create" do
+    subject(:perform_request) do
+      post path, as: request_as, params: {cart_product: {product_id: product.id, quantity: 1}}
+      response
```

Comment:
I am not sure you need to be returning the `response` here. I think it might be returned by the `post` itself and, if it isn't, you can just access `response` directly in the tests anyway (and that's what you do in the test case for `#update`, lines 49 and 63).

Reply:
Hmm, I've always just kind of done this as habit so I can do `it { is_expected.to redirect... }` but if `post` returns the `response` then I guess I don't need to.